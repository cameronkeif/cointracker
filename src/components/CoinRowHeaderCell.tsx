import React from 'react';
import { TableSortDirection, TableSortType } from '../utilities/enums';

export type CoinRowHeaderCellProps = {
  onClick: Function,
  displayText: string,
  tableSortType: TableSortType,
  sortedIcon: JSX.Element | null,
  defaultSortDirection?: TableSortDirection,
};

const CoinRowHeaderCell: React.FC<CoinRowHeaderCellProps> = ({
  onClick,
  displayText,
  tableSortType,
  sortedIcon,
  defaultSortDirection,
}: CoinRowHeaderCellProps) => (
  <th>
    <div
      onKeyDown={(e: React.KeyboardEvent<HTMLSpanElement>) => {
        if (e.key === 'Enter') {
          onClick(tableSortType, defaultSortDirection);
        }
      }}
      onClick={() => onClick(tableSortType, defaultSortDirection)}
      role="button"
      tabIndex={0}
    >
      {displayText}
      {sortedIcon}
    </div>
  </th>
);

CoinRowHeaderCell.defaultProps = {
  defaultSortDirection: TableSortDirection.Ascending,
};

export default CoinRowHeaderCell;
