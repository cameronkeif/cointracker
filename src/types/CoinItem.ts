// A single coin representation
export interface CoinItem {
  symbol: string,
  name: string,
  price: string,
  change: number,
  uuid: string,
  iconUrl: string,
  sparkline: Array<string>,
}

export default CoinItem;
