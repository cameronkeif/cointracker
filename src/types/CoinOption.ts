// Used for the Autocomplete option to be used for coin selection
export interface CoinOption {
  name: string,
  symbol: string,
  uuid: string,
}

export default CoinOption;
