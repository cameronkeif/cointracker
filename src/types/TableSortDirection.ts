export enum TableSortDirection {
  Ascending,
  Descending,
}

export default TableSortDirection;
