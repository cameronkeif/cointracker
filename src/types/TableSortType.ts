export enum TableSortType {
  Name = 'name',
  Symbol = 'symbol',
  Price = 'price',
  Change = 'change',
}

export default TableSortType;
