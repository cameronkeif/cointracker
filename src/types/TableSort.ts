import { TableSortType } from './TableSortType';
import { TableSortDirection } from './TableSortDirection';

export type TableSort = {
  sortType: TableSortType,
  sortDirection: TableSortDirection
};
