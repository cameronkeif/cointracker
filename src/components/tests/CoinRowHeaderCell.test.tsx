import React from 'react';
import ArrowDropUpIcon from '@material-ui/icons/ArrowDropUp';
import { render } from '@testing-library/react';
import CoinRowHeaderCell, { CoinRowHeaderCellProps } from '../CoinRowHeaderCell';
import { TableSortDirection } from '../../types/TableSortDirection';
import { TableSortType } from '../../types/TableSortType';

const defaultProps: CoinRowHeaderCellProps = {
  onClick: jest.fn(),
  displayText: 'name',
  tableSortType: TableSortType.Name,
  sortedIcon: <ArrowDropUpIcon fontSize="small" />,
};

describe('CoinRowHeaderCell', () => {
  it('renders as expected (no sortedIcon)', () => {
    const { asFragment } = render(<CoinRowHeaderCell {...defaultProps} />);
    expect(asFragment).toMatchSnapshot();
  });

  it('renders as expected (ascending)', () => {
    const { asFragment } = render(
      <CoinRowHeaderCell
        {...defaultProps}
      />,
    );
    expect(asFragment).toMatchSnapshot();
  });

  it('renders as expected (descending)', () => {
    const { asFragment } = render(
      <CoinRowHeaderCell
        defaultSortDirection={TableSortDirection.Descending}
        {...defaultProps}
      />,
    );
    expect(asFragment).toMatchSnapshot();
  });
});
