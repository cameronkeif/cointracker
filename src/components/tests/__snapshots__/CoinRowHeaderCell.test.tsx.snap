// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CoinRowHeaderCell renders as expected (ascending) 1`] = `ShallowWrapper {}`;

exports[`CoinRowHeaderCell renders as expected (ascending) 2`] = `ShallowWrapper {}`;

exports[`CoinRowHeaderCell renders as expected (no sortedIcon) 1`] = `ShallowWrapper {}`;
